core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1420,6 +1423,10 @@ class ReplicaManager(val config: KafkaConfig,
           }
           val partitionsToBeFollower = partitionStates.filter { case (k, _) => !partitionsToBeLeader.contains(k) }
 
+          val partitionsToUpdateIdForFollower = topicIdUpdatePartitions.filter { case (_, partitionState) =>
+            partitionState.leader != localBrokerId
+          }
```

Comment:
Could we avoid having to re-iterate over `topicIdUpdatePartitions` by applying the condition before adding the partition to `topicIdUpdatePartitions`?

I have opened a PR to do the same for the two collections above: https://github.com/apache/kafka/pull/11225